frontend/src/routes/Governance/discovered/ByCluster/DiscoveredByCluster.tsx

```diff
@@ -30,50 +41,257 @@ export default function DiscoveredByCluster({
+        deploymentsUnavailable++
+      } else {
+        deploymentsUnknown++
+      }
+
+      if (policy.upgradeAvailable === true) {
+        upgradesAvailable++
+      } else if (policy.upgradeAvailable === false) {
+        upgradesUnavailable++
+      } else {
+        upgradesUnknown++
+      }
+    }
+
+    return {
+      deploymentsAvailable,
+      deploymentsUnavailable,
+      deploymentsUnknown,
+      upgradesAvailable,
+      upgradesUnavailable,
+      upgradesUnknown,
+    }
+  }, [policies, policyKind])
+
+  const filters = useMemo<ITableFilter<DiscoveredPolicyItem>[]>(() => {
+    let filters = [
+      {
+        id: 'violations',
+        label: t('Cluster violations'),
+        options: [
+          {
+            label: t('No violations'),
+            value: 'no-violations',
+          },
+          {
+            label: t('Violations'),
+            value: 'violations',
+          },
+          {
+            label: t('No status'),
+            value: 'no-status',
+          },
+        ],
+        tableFilterFn: (selectedValues: string[], item: DiscoveredPolicyItem): boolean => {
+          let compliant: string
+
+          if (item.apigroup === 'constraints.gatekeeper.sh') {
+            compliant = getConstraintCompliance(item?.totalViolations)
+          } else {
+            compliant = item?.compliant?.toLowerCase() ?? ''
+          }
+
+          for (const value of selectedValues) {
+            if (value === 'no-violations' && compliant === 'compliant') {
+              return true
+            }
+
+            if (value === 'violations' && compliant === 'noncompliant') {
+              return true
+            }
+
+            if (value === 'no-status' && !compliant) {
+              return true
+            }
+          }
+
+          return false
+        },
+      },
+      getResponseActionFilter(t, policies ?? []),
+      getSeverityFilter(t),
+      {
+        id: 'source',
+        label: t('Source'),
+        options: policies ? getSourceFilter(policies) : [],
+        tableFilterFn: (selectedValues: string[], item: DiscoveredPolicyItem) => {
+          if (item.source?.type.toLowerCase() === 'policy') {
+            return selectedValues.includes(item.source.parentNs + '/' + item.source.parentName)
+          }
+
+          return item.source?.type ? selectedValues.includes(item.source?.type) : false
+        },
+      },
+    ] as ITableFilter<DiscoveredPolicyItem>[]
+
+    if (policyKind === 'OperatorPolicy') {
+      filters = filters.concat([
+        {
+          id: 'deploymentAvailable',
+          label: t('Deployment available'),
+          options: [
+            {
+              label: t('yes'),
+              value: 'yes',
+            },
+            {
+              label: t('no'),
+              value: 'no',
+            },
+          ],
+          tableFilterFn: (selectedValues: string[], item: DiscoveredPolicyItem): boolean => {
```

Comment:
value is   // These are only for Operator policy
  deploymentAvailable?: boolean
  upgradeAvailable?: boolean